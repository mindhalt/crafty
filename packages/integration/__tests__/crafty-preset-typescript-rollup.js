/* global describe, it, expect, jest */

const path = require("path");
const rmfr = require("rmfr");
const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

// node-forge 0.6.33 doesn't work with jest.
// but selfsigned is fixed on this version
// as we don't use it for now, we can simply mock it
// https://github.com/jfromaniello/selfsigned/issues/16
jest.mock("node-forge");

it("Works with rollup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-typescript-rollup/compiles"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myTSBundle.min.js")
  ).toMatchSnapshot();
});

it("Deletes rollup uglify plugin using crafty.config.js", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-typescript-rollup/compiles-no-uglify"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js")).toBeTruthy();
  expect(testUtils.exists(cwd, "dist/js/myTSBundle.min.js.map")).toBeTruthy();
  expect(
    testUtils.readForSnapshot(cwd, "dist/js/myTSBundle.min.js")
  ).toMatchSnapshot();
});

it("Fails gracefully on broken markup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-typescript-rollup/fails"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints TypeScript with rollup", async () => {
  const cwd = path.join(
    __dirname,
    "../fixtures/crafty-preset-typescript-rollup/lints"
  );
  await rmfr(path.join(cwd, "dist"));

  const result = await testUtils.run(["run", "default"], cwd);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});
