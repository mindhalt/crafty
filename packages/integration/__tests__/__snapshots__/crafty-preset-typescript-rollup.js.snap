// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deletes rollup uglify plugin using crafty.config.js 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles-no-uglify/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Deletes rollup uglify plugin using crafty.config.js 2`] = `
"var C = /** @class */ (function () {
    function C() {
        /* eslint-disable-next-line no-console */
        console.log(\\"Something happens here\\");
    }
    C.prototype.someMethod = function () {
        /* eslint-disable-next-line no-alert */
        alert(\\"yes, alert still exists\\");
    };
    return C;
}());
var D = /** @class */ (function () {
    function D() {
        /* eslint-disable-next-line no-console */
        console.log(\\"Something else happens here\\");
    }
    D.prototype.someMethod = function () {
        /* eslint-disable-next-line no-alert */
        alert(\\"yes, alert still exists\\");
    };
    return D;
}());

export { C, D };
//# sourceMappingURL=myTSBundle.min.js.map
"
`;

exports[`Fails gracefully on broken markup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/fails/js/typescript.ts
  14:0  error  Parsing error: Unexpected token. A constructor, method, accessor, or property was expected

✖ 1 problem (1 error, 0 warnings)

[js_myTSBundle] (!) Errors were found
js/typescript.ts
[__:__:__] 'js_myTSBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;

exports[`Lints TypeScript with rollup 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/lints/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myBundle' ...

__PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/lints/js/script.ts
   1:1   error    Replace \`⏎import·{\` with \`import·{·\`         @swissquote/swissquote/prettier/prettier
   6:3   error    Replace \`··constructor·\` with \`constructor\`  @swissquote/swissquote/prettier/prettier
   7:5   warning  Unexpected console statement                 no-console
   8:2   error    Insert \`·\`                                   @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                 no-console
  11:17  error    Replace \`·test(·\` with \`test(\`               @swissquote/swissquote/prettier/prettier
  13:16  error    Replace \`·(·)·=>⏎···\` with \`()·=>\`           @swissquote/swissquote/prettier/prettier
  17:7   warning  Unexpected console statement                 no-console
  18:6   error    Replace \`·,·300)\` with \`,·300);\`             @swissquote/swissquote/prettier/prettier

✖ 9 problems (6 errors, 3 warnings)
  6 errors and 0 warnings potentially fixable with the \`--fix\` option.

[js_myBundle] (!) Errors were found
js/script.ts
[__:__:__] 'js_myBundle' errored after ____ ms
[__:__:__] Error: Errors were found
    ...stacktrace...
[__:__:__] 'js' errored after ____ ms
[__:__:__] 'default' errored after ____ ms
",
}
`;

exports[`Works with rollup 1`] = `
Object {
  "status": 0,
  "stdall": "
[__:__:__] Starting Crafty __version__...
[__:__:__] Files will be stored in:
            js: __PATH__/packages/integration/fixtures/crafty-preset-typescript-rollup/compiles/dist/js
[__:__:__] Starting 'default' ...
[__:__:__] Starting 'js' ...
[__:__:__] Starting 'js_myTSBundle' ...
[__:__:__] Finished 'js_myTSBundle' after ____ ms
[__:__:__] Finished 'js' after ____ ms
[__:__:__] Finished 'default' after ____ ms
",
}
`;

exports[`Works with rollup 2`] = `
"var e=/** @class */function(){function e(){console.log(\\"Something happens here\\")}return e.prototype.someMethod=function(){alert(\\"yes, alert still exists\\")},e}(),t=/** @class */function(){function e(){console.log(\\"Something else happens here\\")}return e.prototype.someMethod=function(){alert(\\"yes, alert still exists\\")},e}();export{e as C,t as D};
//# sourceMappingURL=myTSBundle.min.js.map
"
`;
